### PR TITLE
Update Dark ProgramTypes given internal changes and misc work on the editor

### PR DESCRIPTION
**Pull request by OceanOak**
No changelog

This PR:
- Updates Dark ProgramTypes given internal changes
- Fixes package-loading 
- Adds functions to the model in client.dark
- Starts creating a new view where functions, types, handlers, etc will live
- Starts adding a Repl

Current editor state:

https://github.com/darklang/dark/assets/87861924/de837a38-ae19-4665-b37f-f6bce9da6f1b

Follow-ups:
- [ ] Pretty print PipeExpr
- [ ] Editor: When presenting the function list, only show function names without displaying the body.

### CODE REVIEW

File: packages/darklang/prettyPrinter/programTypes.dark [hunk 1]
```diff
@@ -756,8 +865,9 @@ module Darklang =
       let userType
         (userType: PACKAGE.Darklang.Stdlib.ProgramTypes.UserType)
         : String =
+        //don't change userType.name here
```
Comment (OceanOak) [inline packages/darklang/prettyPrinter/programTypes.dark:868]: TODO: remove comment

File: packages/darklang/prettyPrinter/programTypes.dark [hunk 2]
```diff
@@ -783,6 +893,7 @@ module Darklang =
           let typPart =
             PACKAGE.Darklang.PrettyPrinter.ProgramTypes.typeReference p.typ
 
+          // Don't change p.name here it is already a string
```
Comment (OceanOak) [inline packages/darklang/prettyPrinter/programTypes.dark:896]: TODO: remove comment

File: packages/darklang/prettyPrinter/programTypes.dark [hunk 3]
```diff
@@ -22,28 +22,101 @@ module Darklang =
 
             let versionPart =
               if t.version == 0 then "" else $"_v{Int.toString t.version}"
+            // TODO: name should be a string
+            $"{modulesPart}{t.name}{versionPart}"
```
[the remark this answers is not in the document]
Comment (StachuDotNet) [reply packages/darklang/prettyPrinter/programTypes.dark:26]: I think you'll need to pass in a function that toStrings the 'name

For example:
```fsharp
let fullForReference
  (nameToString: 'name -> String)
  (t: PACKAGE.Darklang.Stdlib.ProgramTypes.FQName.BuiltIn<'name>)
  : String =
  let modulesPart =
    match t.modules with
    | [] -> ""
    | modules -> modules |> String.join "." |> (fun parts -> $"{parts}.")

  let versionPart =
    if t.version == 0 then "" else $"_v{Int.toString t.version}"
  $"{modulesPart}{nameToString t.name}{versionPart}"
```

[the remark this answers is not in the document]
Comment (StachuDotNet) [reply packages/darklang/prettyPrinter/programTypes.dark:26]: I hope I'm understanding correctly - but basically I think you might need to
- add this extra argument to the function(s)
- use it (like on the last line)
- supply it where relevant

does that make sense?

**Review by StachuDotNet — APPROVED** [review]
This looks great! I provided feedback on the question around FQName - beyond that, this looks great/merge-able to me.

File: packages/darklang/prettyPrinter/programTypes.dark (hunk 3 above)
[the remark this answers is not in the document]
Comment (OceanOak) [reply packages/darklang/prettyPrinter/programTypes.dark]: Yes, it does. Thank you!

**Pull request merged**